bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -28,215 +28,150 @@ spec:
-
-      # run-ovn-northd - v3
-      - name: run-ovn-northd
+      - name: northd
         image: {{.OvnImage}}
-
-        command: ["/root/ovnkube.sh", "run-ovn-northd"]
-
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f /env/_master ]]; then
+            set -o allexport
+            source /env/_master
+            set +o allexport
+          fi
+          exec ovn-northd --no-chdir "-vconsole:${OVN_LOG_LEVEL}" -vfile:off --pidfile=/var/run/openvswitch/ovn-northd.pid
+        env:
+        - name: OVN_LOG_LEVEL
+          value: info
         volumeMounts:
-        # ovn db is stored in the pod in /etc/openvswitch (default ovs tool location)
-        # and on the host in /var/lib/openvswitch/
         - mountPath: /etc/openvswitch/
-          name: host-var-lib-ovs
+          name: etc-openvswitch
         - mountPath: /var/lib/openvswitch/
-          name: host-var-lib-ovs
-        - mountPath: /var/run/openvswitch/
-          name: host-var-run-ovs
-
+          name: var-lib-openvswitch
+        - mountPath: /run/openvswitch/
+          name: run-openvswitch
+        - mountPath: /env
+          name: env-overrides
         resources:
           requests:
             cpu: 100m
             memory: 300Mi
-        env:
-        - name: OVN_DAEMONSET_VERSION
-          value: "3"
-        - name: OVN_LOG_NORTHD
-          value: "-vconsole:info"
-        - name: OVN_NB_PORT
-          value: "9641"
-        - name: OVN_SB_PORT
-          value: "9642"
-        - name: OVN_NET_CIDR
-          valueFrom:
-            configMapKeyRef:
-              name: ovn-config
-              key: net_cidr
-        - name: OVN_SVC_CIDR
-          valueFrom:
-            configMapKeyRef:
-              name: ovn-config
-              key: svc_cidr
-        - name: K8S_NODE
-          valueFrom:
-            fieldRef:
-              fieldPath: spec.nodeName
-        - name: K8S_APISERVER
-          valueFrom:
-            configMapKeyRef:
-              name: ovn-config
-              key: k8s_apiserver
-        - name: OVN_KUBERNETES_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
-      # end of container
 
-      # nb-ovsdb - v3
-      - name: nb-ovsdb
+      - name: nbdb
         image: {{.OvnImage}}
-
-        command: ["/root/ovnkube.sh", "nb-ovsdb"]
-
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f /env/_master ]]; then
+            set -o allexport
+            source /env/_master
+            set +o allexport
+          fi
+          exec /usr/share/openvswitch/scripts/ovn-ctl run_nb_ovsdb \
+            --no-monitor \
+            --ovn-nb-log="-vconsole:${OVN_LOG_LEVEL} -vfile:off"
+        lifecycle:
+          postStart:
+            exec:
+              command: ["ovn-nbctl", "set-connection", "ptcp:{{.OVN_NB_PORT}}", "--", "set", "connection", ".", "inactivity_probe=0"]
```

Comment:
Is it guaranteed that nbdb will be ready by the earliest time kubelet could decide to run this command?

Reply:
Nope, and even worse: it's not guaranteed that postStart hooks will be run after the container's entrypoint is executed! I'll add a loop.